markdown.dtx

```diff
@@ -30959,17 +30995,44 @@ M.extensions.fenced_divs = function(blank_before_div_fence)
 %
 % If the `blank_before_div_fence` parameter is `false`, we will have the
 % closing div at the beginning of a line break the current paragraph if
-% we are currently nested in a div.
+% we are currently nested in a div and the indentat matches the opening
+% div fence.
 %
 % \end{markdown}
 %  \begin{macrocode}
-      local function check_div_level(s, i, current_level) -- luacheck: ignore s i
-        current_level = tonumber(current_level)
-        return current_level > 0
+      local function is_inside_div()
+        local function check_div_level(s, i, fenced_div_level) -- luacheck: ignore s i
+          fenced_div_level = tonumber(fenced_div_level)
+          return fenced_div_level > 0
+        end
+
+        return Cmt(Cb("fenced_div_level"), check_div_level)
+      end
+
+      local function check_indent()
+        local function compare_indent(s, i, indent_table, -- luacheck: ignore s i
+                                      fenced_div_indent_table, fenced_div_level)
+          fenced_div_level = tonumber(fenced_div_level)
+          local num_current_indents = 0
+          local num_opening_indents = 0
+          if indent_table.indents ~= nil then
+            num_current_indents = #indent_table.indents
+          end
+          fenced_div_indent_table = fenced_div_indent_table[fenced_div_level]
+          if fenced_div_indent_table.indents ~= nil then
+            num_opening_indents = #fenced_div_indent_table.indents
+          end
+          return num_current_indents == num_opening_indents
```

Comment:
@lostenderman If you are busy, I will likely figure it out eventually at this point. However, you would likely be much more efficient at this, since the indent-info / trail-info objects are your code.